### ISSUE TITLE
Can x axis be configured to hide some of the labels for large data sets?

### ISSUE THREAD

**Issue by stryecho** [tomauty/react-native-chart#98]
Is there a way to hide some of the x labels so that they don't get clipped?  I'm using a test case with a couple hundred data points like so:

```
    let data = [
       [0, 1], [0, 1]...repeat a couple hundred times...[0, 1]
    ];
    resultsBody = <Chart style={styles.chart}
                    data={dataPoints}
                    verticalGridStep={7}
                    type='line'
                    yAxisWidth={40}
                    showGrid={false}
                  />
```

When I render the chart, the lines are all clipped, as expected:

![toomanylabels](https://cloud.githubusercontent.com/assets/5504328/16570451/98cb19ce-4202-11e6-8026-fd184d00952f.png)

I'm looking for a way to show just every nth label.  In the previous (1.1.4) version, I was able to get around the issue by specifying an empty string for labels I did not want to present.  It would then render only the labels that were defined.  However, in the current 1.0.1-beta version, when I render the chart, the empty labels still seem to affect the defined labels by clipping them:

![setsomeemptylabels](https://cloud.githubusercontent.com/assets/5504328/16570480/275993a0-4203-11e6-9b1e-ebdf811e894f.png)

**Comment by tomauty** [comment]
Would the previous solution be reasonable for you? Or do you think it would contribute more to actually have some kind of prop to handle this case?

**Comment by stryecho** [comment]
For a quick solution, I think that the previous solution is fine (which is why I went with it in the 1.1.4 version).  Having a prop that specifies which label to show (may be similar to issue #46, kind of like step from HighCharts (http://api.highcharts.com/highcharts#xAxis.labels.step)) would be much more convenient as the user wouldn't need to change the data array which leads me to another, related, observation.

This may be another feature request (I think related to issue #62 about non constant x,y step), but specific tot he label issue.  Because the x axis data point is also used as a label, the "empty" x data point solution doesn't work well for points that are inconsistently spaced.  In that scenario, I think it would be better to specify x,y data points and have a separate set of labels.

For example, if we have 

```
let data = [
   [0, 1], [7, 1], [8, 1], [22,1], [24,1]...etc, etc....[201, 1]
];
```

And we use the simple case of just setting x to "empty" and we display every second label, the array would then look like:

```
let data = [
   [0, 1], ['', 1], [8, 1], ['',1], [24,1]...etc, etc....[201, 1]
];
```

In this case, RNCharts would not know where to place the [7,1] data point since it's x value is empty.  I think right now RNCharts just consistently spaces all points and just uses the x value as a label so at the moment it's irrelevant.  However, if we were to ever want to support data points spaced differently than the axis, we'd need to keep the data points and then have a separate set of labels.

I'm not too familiar with how the rendering is handled, but I would suspect that we would need to have data like so:

```
let data = [
   [0, 1], [7, 1], [8, 1], [22,1], [24,1]...etc, etc....[201, 1]
];
let labels = [
   0, 50, 100, 150, 200, 250
]
```

And then render the data points against the labels on the x-axis.

TL;DR  If you don't intend to support inconsistently spaced x-axis data points (but have a consistently spaced set of labels), having an "empty" array is good and a prop for "step" would be better.  If you want to eventually support inconsistently spaced x-axis points, then having a separate label property would be necessary.

**Comment by tomauty** [comment]
Yeah, the previous version made `xLabels` mandatory. I took that out for simplicity's sake, but might make it optional. I'll have to think it over for a bit. Thanks for the help! Lemme know if you have any additional thoughts.

**Comment by stryecho** [comment]
Thanks for your work on the project!

Do you think you might implement either the empty label or step solution soon?  I'm looking at upgrading react-native and will need to upgrade RNCharts to work with it (because of the change in 26 that now requires the react api to be pulled from react).

**Comment by tomauty** [comment]
I'll start by making it the case that empty strings don't get rendered as elements. Should be out today.

**Comment by stryecho** [comment]
Thanks.

**Comment by tomauty** [comment]
`1.0.6-beta` SHOULD work for this :)
